### PR TITLE
Private app support

### PR DESCRIPTION
**Pull request by airhorns**
This is on top of #16 to add support for private Shopify apps via a disjoint set of config options. I addressed the remaining feedback from that PR but feel free to add more!

### CODE REVIEW

**Comment by cmerrick** [comment]
Hi @airhorns, thanks for your contribution!

In order for us to evaluate and accept your PR, we ask that you [sign a contribution license agreement](https://stitch-cla-enforcer.herokuapp.com/). It's all electronic and will take just minutes.

**Comment by cmerrick** [comment]
You did it @airhorns!

Thank you for signing the Singer Contribution License Agreement.

**Comment by airhorns** [comment]
@timvisher would you be able to take a peek at this as you were the last reviewer of #16 ?

**Comment by KAllan357** [comment]
@airhorns could you fix the pylint issues indicated here: https://circleci.com/gh/singer-io/tap-shopify/501

**Comment by airhorns** [comment]
Whoops! Fixed.

**Comment by airhorns** [comment]
@KAllan357 I'm not really sure how to debug those tests that use `target-stitch` but I'm happy to if you give me some pointers! Looking at the logs it seems like something somewhere is trying to execute an env var as a command:

```
Exception: check invocation failed. command: $STITCH_ORCHESTRATOR_PATH
```

and, there seems to be an issue with `target-stitch` complaining about an extraneous key:

```
Exception: Error saving discovered stream transactions: {"status":"BadRequest","message":"Invalid streams specification","errors":["tap_name : disallowed-key"]}
```

Any pointers or advice for fixing this failure ?

**Comment by dmosorast** [comment]
@airhorns Sorry about that, that was due to a change I made yesterday and didn't rebuild the test environment, I'll re-trigger the build in a few minutes.

File: tap_shopify/__init__.py [hunk 1]
```diff
@@ -14,14 +14,35 @@
...
+        shop_url = 'https://{}:{}@{}/admin'.format(api_key, password, shop)
+
+        shopify.ShopifyResource.set_site(shop_url)
+    else:
+        raise Exception(
+            "Config is missing Shopify auth information." +
+            "Either pass the `api_key` option for a public app authenticated using an" +
+            " access_token via the Shopify OAuth flow, or pass the `private_app_api_key`" +
+            " and `private_app_password` config options for a private app."
+        )
```
Comment (luandy64) [inline tap_shopify/__init__.py]: This worked with `pylint` locally and is how I like to do multiline strings in python
```
        raise Exception(
            ("Config is missing Shopify auth information. Either pass the `api_key` option for a "
             "public app authenticated using an access_token via the Shopify OAuth flow, or pass "
             "the `private_app_api_key` and `private_app_password` config options for a private "
             "app."))
```

**Comment by airhorns** [comment]
Ok, this is updated and should be good to go!

**Comment by dougb** [comment]
@airhorns Is there any way I can help you on this ? I need it to fix #47

**Comment by airhorns** [comment]
@dougb I'm actually waiting on the maintainers to merge, I've addressed the feedback, maybe @luandy64 has more concerns though!

**Comment by DouweM** [comment]
It appears that `tap-shopify` already supports private apps without these changes, since the `X-Shopify-Access-Token` request header that is set to the value of the `api_key` config doesn't just support OAuth access tokens, but also private app API passwords per [the docs](https://shopify.dev/tutorials/authenticate-a-private-app-with-shopify-admin#make-authenticated-requests)!

**Comment by airhorns** [comment]
@DouweM thats good to know! I feel like it might be a bit more obvious to have formal support for it in the docs/config so it's explicit what's going on but if this is a wontmerge I understand.

**Comment by DouweM** [comment]
@airhorns I agree it should be made clear in the README that `api_key` can be an OAuth access token _or_ a private app API password, but I was glad to find that no actual changes to the code are necessary :)

**Comment by airhorns** [comment]
@DouweM thats good to know! I feel like it might be a bit more obvious to have formal support for it in the docs/config so it's explicit what's going on but if this is a wontmerge I understand.

**Pull request closed without merging**